src/components/ProductsList/ProductsListItem.tsx

```diff
@@ -0,0 +1,76 @@
+import React, { useState } from 'react';
+import {
+  IconButton,
+  ListItem,
+  ListItemSecondaryAction,
+  ListItemText,
+} from '@material-ui/core';
+import './ProductsListItem.scss';
+import { RemoveModal } from 'components/RemoveModal';
+import EditIcon from '@material-ui/icons/Edit';
+import DeleteIcon from '@material-ui/icons/Delete';
+import { Product as ProductType } from 'types/product';
+import { ProductsListEditForm } from './ProductsListEditForm';
+
+type ProductProps = {
+  product: ProductType;
+  onDelete: () => Promise<void>;
+  onEdit: ({ id, name }: ProductType) => Promise<void>;
+  isEditPending: boolean;
+};
+
+type ViewMode = 'read' | 'edit';
+
+export const ProductsListItem: React.FC<ProductProps> = ({
+  product,
+  onDelete,
+  onEdit,
+  isEditPending,
+}) => {
+  const [mode, setMode] = useState<ViewMode>('read');
+  const [isModalOpen, setIsModalOpen] = useState<boolean>(false);
+
+  return (
+    <div
+      className={`product-container ${
```

Comment:
What about use `clsx`?
```suggestion
      className={clsx('product-container', isEditPending && 'product-container--loading')}
```